lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift

```diff
@@ -0,0 +1,312 @@
+            // A regex matching that matches the generic type.
+            typeIdentifier = "^\(moduleName)\\.\(typeName)<\(typePatterns)>"
+        } else if declaration.is(ExtensionDeclSyntax.self) {
+            // When attached to an extension, the type may or may not be a generic type.
+            // This regular expression handles both cases.
+            typeIdentifier = "^\(moduleName)\\.\(typeName)(<.+>)?$"
+        } else {
+            typeIdentifier = "\(moduleName).\(typeName)"
+        }
+
+        // Serialize the type summary into a global record, in a custom section, for LLDB to load.
+        let decl: DeclSyntax = """
+        #if os(Linux)
+        @_section(".lldbsummaries")
+        #elseif os(Windows)
+        @_section(".lldbsummaries")
+        #else
+        @_section("__DATA_CONST,__lldbsummaries")
+        #endif
+        @_used
+        let \(raw: typeName)_lldb_summary = (
+            \(raw: encodeTypeSummaryRecord(typeIdentifier, summaryString))
+        )
+        """
+
+        return [decl]
+    }
+}
+
+// MARK: - Diagnostics
+
+private struct ErrorMessage: DiagnosticMessage, ExpressibleByStringInterpolation {
+    init(stringLiteral value: String) {
+        self.message = value
+    }
+    var message: String
+    var diagnosticID: MessageID { .init(domain: "DebugDescription", id: "DebugDescription")}
+    var severity: DiagnosticSeverity { .error }
+}
+
+extension MacroExpansionContext {
+    fileprivate func diagnose(node: some SyntaxProtocol, message: any DiagnosticMessage) {
+        diagnose(Diagnostic(node: node, message: message))
+    }
+}
+
+// MARK: - AST Helpers
+
+private func extractModuleName(from fileID: ExprSyntax) -> String? {
+    if let fileID = fileID.as(StringLiteralExprSyntax.self)?.representedLiteralValue,
+       let firstSlash = fileID.firstIndex(of: "/") {
+        return String(fileID.prefix(upTo: firstSlash))
+    }
+    return nil
+}
+
+private func identifier<Decl: DeclSyntaxProtocol>(of decl: Decl) -> TokenSyntax? {
+    decl.as(StructDeclSyntax.self)?.name ??
+    decl.as(ClassDeclSyntax.self)?.name ??
+    decl.as(EnumDeclSyntax.self)?.name ??
+    decl.as(ExtensionDeclSyntax.self)?.extendedType.as(IdentifierTypeSyntax.self)?.name
+}
+
+private func memberBlock<Decl: DeclSyntaxProtocol>(_ decl: Decl) -> MemberBlockSyntax? {
+    decl.as(StructDeclSyntax.self)?.memberBlock ??
+    decl.as(ClassDeclSyntax.self)?.memberBlock ??
+    decl.as(EnumDeclSyntax.self)?.memberBlock ??
+    decl.as(ExtensionDeclSyntax.self)?.memberBlock
+}
+
+private func syntaxKeyword<Decl: DeclSyntaxProtocol>(for decl: Decl) -> String {
+    if decl.is(StructDeclSyntax.self) { return "struct" }
+    if decl.is(ClassDeclSyntax.self) { return "class" }
+    if decl.is(EnumDeclSyntax.self) { return "enum" }
+    if decl.is(ExtensionDeclSyntax.self) { return "extension" }
+    assertionFailure("expected struct/class/enum/extension")
+    return "declaration"
```

Comment:
A nicer way to do this, IMO, is 

```swift
switch decl.syntaxKind {
case .struct: return "struct"
}
```

---

You should support actors here as well.